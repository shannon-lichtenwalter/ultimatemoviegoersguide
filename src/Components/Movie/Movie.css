.poster {
  height: 300px;
  width: 200px;
  background-color: lightgray;
  display:flex;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
}

.posterImage {
  border-radius: 5px;
}

.poster:hover, .posterImage:hover {
  opacity: 0.6;
  
}

.movieListItem:hover {
  background-color: rgb(237, 235, 231, 0.6);
}