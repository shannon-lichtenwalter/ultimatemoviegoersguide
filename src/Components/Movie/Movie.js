import React from 'react';
import './Movie.css';


class Movie extends React.Component{
  renderMovieDetails = () => {
    this.props.showDetails(this.props.data)
  }

  render(){
    return(
      <li onClick={()=> this.renderMovieDetails()}>
        {this.props.data.poster_path 
          ? <img src={`https://image.tmdb.org/t/p/w200/${this.props.data.poster_path}`} alt='movie poster'/>
          : <div className='poster'>No Poster Data Available</div>
        }
        <h1>{this.props.data.title}</h1>
      </li>
    )
  }
}

export default Movie;