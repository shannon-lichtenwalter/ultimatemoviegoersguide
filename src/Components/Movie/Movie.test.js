import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Movie from './Movie';

describe('Movie Component', () => {
  const testData = {
    id: '123',
    poster_path: '/test',
    title: 'test'
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<Movie data={testData}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  }); 

  it('Displays the Movie component correctly when rendered', () => {
    const wrapper = mount(<Movie data={testData}/>);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});
