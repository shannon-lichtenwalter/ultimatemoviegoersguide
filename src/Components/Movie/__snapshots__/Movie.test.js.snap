// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component Displays the Movie component correctly when rendered 1`] = `
<Movie
  data={
    Object {
      "id": "123",
      "poster_path": "/test",
      "title": "test",
    }
  }
>
  <li
    className="movieListItem"
  >
    <button
      className="movieButton"
      onClick={[Function]}
    >
      <img
        alt="movie poster"
        className="posterImage"
        src="https://image.tmdb.org/t/p/w200//test"
      />
      <h1>
        test
      </h1>
    </button>
  </li>
</Movie>
`;
